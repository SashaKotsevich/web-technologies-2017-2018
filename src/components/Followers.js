import React from 'react';
import '../index.css';
import {connect} from 'react-redux';
import request from '../api.js';
class FollowersPart extends React.Component{

    reload=(event,count)=>{
        this.props.FollowersRequest(document.getElementById("input_area").value);
        let text=' ';
        this.props.state.FollowersReducer.map(item=>{
                text=text+item+'\n';
        })
       
        document.getElementById("ShowFollovers").value=text;
      
    }
   

    render(){
        
        return(
             <div>
                 <button onClick={this.reload}>Load</button>
                 <text>Пока по кнопке нужно нажать пару раз связано с временем запроса потом поправлю :)</text>
                <textarea id="ShowFollovers"  className="ScrollPanel"></textarea>
             </div>

        )
    }

}



export default connect(
    state=>({
      state
          }),
   

    dispatch =>({
        FollowersRequest: (user) =>{
            request("https://api.github.com/users/"+user+"/followers")
                    .then((user) => {
                        dispatch({
                            type: 'Follow_Rqst',
                            data: user
                        })
                    })
                    .catch((e) => {
                        dispatch({type: 'user_not_found'})
                    })
        }
    })
  )(FollowersPart);