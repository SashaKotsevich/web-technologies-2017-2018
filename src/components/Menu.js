import React from 'react';
import { Link } from 'react-router-dom';


class Menu extends React.Component{
    render(){
        return(
            <div>
            <Link to="/">Tabs</Link>
            <Link to="/followers">Followers</Link>
            </div>
        )

    }
}

export default Menu;