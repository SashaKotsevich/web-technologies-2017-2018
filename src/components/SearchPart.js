import React from 'react';
import '../index.css';
import {connect} from 'react-redux';
import * as actions from "../actions/actions"



export class Icon extends React.Component{
    render(){
        return(
        <i className={this.props.clname} aria-hidden="true"></i>
        );
    }
}

 class SearchInfo extends React.Component{
    render(){
        return(
            <div>
                
                <img className="avatar_url" alt="" src={this.props.data.User.avatar_url }></img>
                <p className="name">{this.props.data.User.name}</p>
                <p className="login">{this.props.data.User.login}</p>
                <p className="bio">{this.props.data.User.bio}</p>
                <hr className="hr_line"/>
                <div className="company_line"> <Icon clname="fa fa-users"/> <p className="company">{this.props.data.User.company}</p></div>
                <div className="marker_line"><Icon clname="fa fa-map-marker "/><p className="location">{this.props.data.User.location}</p></div>
                <div className="email_line"><Icon clname="fa fa-envelope" /><a className="email" src={this.props.data.User.email}>{this.props.data.User.email}</a></div>
                <div className="blog_line"><con clname="fa fa-retweet" /><a className="blog" src={this.props.data.User.blog}>{this.props.data.User.blog}</a></div>
            </div>
        );
    }
}




 class SearchPart extends React.Component{
  
    handlesearch= (event)=>{
       //get userinfo

        let searchitem=document.getElementById("UserNameTextArea").value;
        if(searchitem!==""){
        this.props.changeuser("https://api.github.com/users/"+searchitem);
         this.props.changefollows("https://api.github.com/users/"+searchitem+'/followers');
         this.props.changeorgs("https://api.github.com/users/"+searchitem+'/orgs');
         this.props.changerepos("https://api.github.com/users/"+searchitem+'/repos');
         this.props.changetoppopular('https://api.github.com/search/repositories?q=stars%3A%3E0&sort=stars&per_page=5');
         this.props.changetopgrowing('https://api.github.com/search/repositories?q=trending%3A%3E0&sort=stars&per_page=5&since=weekly&#39');
        }
        else{alert("Заполните поле поиска")}

   
    }


    render(){
        return(
            <div className="search_user_window">
                <input type="text" id="UserNameTextArea"  />
                <input type="button" value="Search" className="button_search" onClick={this.handlesearch}/>
                               
                <SearchInfo data={this.props.state} />
                
            </div>
        )
    }
}

  export default connect(
    state=>({
      state
          }),
   

    dispatch =>({
        changeuser: (link) =>{
            dispatch(actions.changeuseraction(link))
        },
        changerepos: (link) =>{
            dispatch(actions.changereposaction(link))
        },
        changefollows: (link) =>{
            dispatch(actions.changefollowsaction(link))
        },
        changeorgs: (link) =>{
            dispatch(actions.changeorgsaction(link))
        },
        changetab: (data) =>{
            dispatch(actions.changetabaction(data))
        },
        changetoppopular: (link) =>{
            dispatch(actions.changetoppopularaction(link))

        },
        changetopgrowing: (link) =>{
            dispatch(actions.changetopgrowingaction(link))

        }

        
    })
    
  )(SearchPart);