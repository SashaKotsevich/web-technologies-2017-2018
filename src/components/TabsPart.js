import React from 'react';
import '../index.css';
import {connect} from 'react-redux';

 class TabsPart extends React.Component{
       
    switchtab= (event)=>{
        document.getElementById("first_tab").className="button_emphasis_none";
        document.getElementById("second_tab").className="button_emphasis_none";
        document.getElementById("third_tab").className="button_emphasis_none";
        document.getElementById(event.target.className).className="button_emphasis";
        this.props.changetab(event.target.className,this.props.state[event.target.className])
        document.getElementById("TextArea").value=this.props.state[this.props.state.currenttab];
        
    }

   handletab=(event)=>{
     
        this.props.changetab(this.props.state.currenttab,event.target.value);
      
    }
   textediting=(event)=>{
      document.getElementById("TextArea").disabled=!document.getElementById("TextArea").disabled;
       
    }
    render(){
               return(
            <div className="tabs_part">
                <div className="button_panel">
                    <div className="button_block"> <input type="button" value="Основное" className="first_tab" onClick={this.switchtab}/> 
                    <hr className="button_emphasis" id="first_tab"/>
                     </div>
                    <div className="button_block"><input type="button" value="Образование" className="second_tab" onClick={this.switchtab}/>
                     <hr className="button_emphasis_none" id="second_tab"/> 
                     </div>
                    <div className="button_block"><input type="button" value="Контакты" className="third_tab" onClick={this.switchtab}/>
                     <hr className="button_emphasis_none" id="third_tab"/>
                      </div>
                    <input type="button" value="edit" className="button_edit" onClick={this.textediting} /> 
                </div>
                
                {  <textarea className="tab_text" id="TextArea" disabled onChange={this.handletab}/> } 
            </div>
        )
    }

}

export default connect(
    state=>({
     state:state.TabsReducer
              
    }),
    dispatch =>({
        changetab: (cu,text) =>{
            dispatch({type:"ChangeTaCt", ct:cu , txt:text})
        }
    })
  )(TabsPart);