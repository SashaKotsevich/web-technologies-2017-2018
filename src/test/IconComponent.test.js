import {Icon} from "../components/SearchPart";
import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';  
configure({adapter: new Adapter()});
describe('Icon component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Icon clname="ttp"/>)
    });
    it('Icon Snapshot', () => {
        const rendered = renderer.create(<Icon clname="ttp"/>).toJSON();
        expect(rendered).toMatchSnapshot();
    });
    it('Icon Render', () => {
        expect(wrapper.length).toEqual(1);
    });
});
