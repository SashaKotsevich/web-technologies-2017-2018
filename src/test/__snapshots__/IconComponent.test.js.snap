// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Icon Snapshot 1`] = `
<i
  aria-hidden="true"
  className="ttp"
/>
`;
